### PR TITLE
Tooling upgrades

### PR DESCRIPTION
**Pull request by jodal**
- **Require Python 3.11+**
- **Replace pkg_resources with importlib.metadata**
- **Move project declaration to pyproject.toml**
- **Change project layout**
- **Replace flake8 and black with ruff**
- **Format with ruff**
- **Autofix with ruff**
- **Autofix with ruff --unsafe-fixes**
- **Manually fix ruff lints**

### CODE REVIEW

**Comment by jodal** [comment]
Rebased on top of today's release now, and added the version simplification related to mopidy/mopidy#2153.

File: pyproject.toml [hunk 1]
```diff
@@ -1,17 +1,111 @@
...
-[tool.black]
-target-version = ["py39", "py310", "py311"]
-line-length = 80
+[project]
+name = "Mopidy-Spotify"
+description = "Mopidy extension for playing music from Spotify"
+readme = "README.rst"
+requires-python = ">= 3.11"
+license = { text = "Apache-2.0" }
+authors = [{ name = "Stein Magnus Jodal", email = "stein.magnus@jodal.no" }]
```
Comment (kingosticks) [inline pyproject.toml:12]: Maybe we need a `maintainers` key. But I am not bothered, you can forward me the 2 emails that come through!

File: src/mopidy_spotify/web.py [hunk 2]
```diff
@@ -321,10 +325,12 @@ def _parse_etag(response):
             # Format is string of ASCII characters placed between double quotes
             # but can seemingly also include hyphen characters.
             etag = re.match(r'^(W/)?("[!#-~]+")$', value)
-            if etag and len(etag.groups()) == 2:
+            if etag and len(etag.groups()) == 2:  # noqa: PLR2004
                 return etag.groups()[1]
+            return None
```
Comment (kingosticks) [inline src/mopidy_spotify/web.py:330]:
```suggestion
```

File: src/mopidy_spotify/web.py [hunk 3]
```diff
@@ -568,22 +567,18 @@ def from_uri(cls, uri):
...
-                return cls(uri, LinkType.PLAYLIST, None, parts[1])
-        elif len(parts) == 3 and parts[0] == "playlist":
-            return cls(uri, LinkType.PLAYLIST, parts[2], parts[1])
-        elif len(parts) == 4 and parts[0] == "user" and parts[2] == "playlist":
-            return cls(uri, LinkType.PLAYLIST, parts[3], parts[1])
+        match parts:
+            case [type, id] if type in ("track", "album", "artist", "playlist"):
+                return cls(uri, LinkType(type), id, None)
+            case ["your", _]:
+                return cls(uri, LinkType("your"))
```
Comment (kingosticks) [inline src/mopidy_spotify/web.py:574]:
```suggestion
                return cls(uri, LinkType.YOUR)
```

**Review by kingosticks — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (jodal) [reply pyproject.toml:12]: There is support for it! https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#authors-maintainers

**Pull request merged**